crates/consensus/src/transaction/envelope.rs

```diff
@@ -448,16 +448,21 @@ impl OpTxEnvelope {
     }
 
     /// Returns the inner transaction hash.
-    pub fn tx_hash(&self) -> B256 {
+    pub fn hash(&self) -> &B256 {
         match self {
-            Self::Legacy(tx) => *tx.hash(),
-            Self::Eip1559(tx) => *tx.hash(),
-            Self::Eip2930(tx) => *tx.hash(),
-            Self::Eip7702(tx) => *tx.hash(),
-            Self::Deposit(tx) => tx.tx_hash(),
+            Self::Legacy(tx) => tx.hash(),
+            Self::Eip1559(tx) => tx.hash(),
+            Self::Eip2930(tx) => tx.hash(),
+            Self::Eip7702(tx) => tx.hash(),
+            Self::Deposit(tx) => tx.hash_ref(),
         }
     }
 
+    /// Returns the inner transaction hash.
+    pub fn tx_hash(&self) -> B256 {
```

Comment:
is more straight forward if this method is called `hash_copied`, alt `hash` and call the other one `hash_ref`